llvm/unittests/ADT/ArrayRefTest.cpp

```diff
@@ -255,6 +255,26 @@ TEST(ArrayRefTest, ArrayRefFromStdArray) {
   }
 }
 
+TEST(ArrayRefTest, ArrayRefFromIteratorRange) {
+  std::array<int, 5> A1{{42, -5, 0, 1000000, -1000000}};
+  ArrayRef<int> A2 = make_range(A1.begin(), A1.end());
+
+  EXPECT_EQ(A1.size(), A2.size());
+  for (std::size_t i = 0; i < A1.size(); ++i) {
```

Comment:
nit: You can use `EXPECT_THAT(A2, ElementsAreArray(A1))`

Reply:
I could 't get that to work, due to some template errors. I didn't spend too much time to figure out why it's not working and left it as-is for now

I also left the existing loop instead of a range based for loop, as LLVM's `zip` doesn't seem to work nicely with std::array.